### PR TITLE
Add `recruitment/create-requisition` profile

### PR DESCRIPTION
**Pull request by janhalama**
## Description

This PR adds `recruitment/create-requisition` profile along with `mock` and `lever` provider maps.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: providers/lever.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "lever",
+  "services": [
+    {
+      "id": "default",
+      "baseUrl": "https://{DOMAIN}.lever.co"
```
Comment (tmladek) [inline providers/lever.json:6]: nitpick: consider using `SUBDOMAIN`, for consistency (?)

File: grid/recruitment/create-requisition/profile.supr [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+   """
+   On hold job requisition
+   Pending requisitions are deferred, recruiters are not actively looking to fill the position.
+   """
+   OnHold
+}
+
+"""
+Employment Status
+The employment classification.
```
Comment (tmladek) [inline grid/recruitment/create-requisition/profile.supr:172]:
```suggestion
The employment status of a worker, such as part-time, full-time, contractor, etc.
```

File: grid/recruitment/create-requisition/profile.supr [hunk 3]
```diff
@@ -0,0 +1,195 @@
+"""
+Create Requisition
+Create new requisition in ATS (Applicant Tracking System).
+"""
+name = "recruitment/create-requisition"
+version = "1.0.0"
+
+"""
+Create Requisition
+Creates job requisition to fill job vacancy in your company.
```
Comment (tmladek) [inline grid/recruitment/create-requisition/profile.supr:10]:
```suggestion
Creates a job requisition to fill a job vacancy in a company.
```

File: grid/recruitment/create-requisition/profile.supr [hunk 4]
```diff
@@ -0,0 +1,195 @@
+"""
+Create Requisition
+Create new requisition in ATS (Applicant Tracking System).
```
Comment (tmladek) [inline grid/recruitment/create-requisition/profile.supr:3]:
```suggestion
Create a new requisition in an ATS (Applicant Tracking System).
```

File: grid/recruitment/create-requisition/maps/mock.test.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+import { buildSuperfaceTest } from '../../../test-config';
+
+describe('recruitment/create-requisition/mock', () => {
+  let superface: SuperfaceTest;
+
+  beforeEach(() => {
+    superface = buildSuperfaceTest({
+      profile: 'recruitment/create-requisition',
+      provider: 'mock',
```
Comment (tmladek) [inline grid/recruitment/create-requisition/maps/mock.test.ts:11]: Why make a special test for mock? A lot of other profiles in station simply use the same generic test.

File: grid/recruitment/create-requisition/maps/create-requisition.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            requisitionCode: testInputs.existingRequisitionCode,
+            name: 'Software Developer, Platform',
+            headcountTotal: 2,
+            status: 'Open',
+            location: 'Prague',
+          },
+        });
+
+        expect(() => result.unwrap()).toThrow();
+        expect(result).toMatchSnapshot();
```
Comment (tmladek) [inline grid/recruitment/create-requisition/maps/create-requisition.ts:53]: I suggest checking the error type returned, as in e.g. https://github.com/superfaceai/station/blob/main/grid/recruitment/create-candidate/maps/create-candidate.ts#L189

(Or preferably a prettier way, if you find it 😅 )

File: grid/recruitment/create-requisition/maps/lever.suma [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        case 'FullTime':
+          return 'full-time';
+        case 'PartTime':
+          return 'part-time';
+        case 'contractor':
+          return 'contractor';
+        case 'TempWorker':
+          return 'temp-worker'
+        default:
+          return 'full-time';
```
Comment (tmladek) [inline grid/recruitment/create-requisition/maps/lever.suma:141]: Should we have a default for this input attribute? (I'd at least document it somewhere)

**Review by tmladek — CHANGES_REQUESTED** [review]

File: grid/recruitment/create-requisition/maps/lever.suma (hunk 7 above)
Comment (janhalama) [reply grid/recruitment/create-requisition/maps/lever.suma]: 👍 I made the `employmentStatus` required in profile.

**Comment by janhalama** [comment]
@tmladek is it still worth finishing this task? Do you have capacity to finish the review or should I close it?

**Comment by janhalama** [comment]
We no longer have any demand for the 'recruitment/create-requisition' profile.

**Pull request closed without merging**